### PR TITLE
add tyr templates for debian8

### PR DESCRIPTION
**Pull request by Hoshiyo**

### CODE REVIEW

File: fabfile/component/tyr.py [hunk 1]
```diff
@@ -118,8 +118,13 @@ def setup_tyr():
 
     update_tyr_conf()
 
-    _upload_template('tyr/tyr_worker.jinja', env.tyr_worker_service_file,
+    if env.distrib == 'debian8':
```
Comment (antoine-de) [inline fabfile/component/tyr.py]: can't we have a env.use_systemd ?

**Comment by antoine-de** [comment]
docker doesn't seems to like this

**Comment by Hoshiyo** [comment]
yes I forget that

**Comment by Hoshiyo** [comment]
@antoine-de :)

File: fabfile/component/tyr.py [hunk 2]
```diff
@@ -118,8 +118,13 @@ def setup_tyr():
 
     update_tyr_conf()
 
-    _upload_template('tyr/tyr_worker.jinja', env.tyr_worker_service_file,
+    if env.distrib == 'debian8' and env.use_systemd:
```
Comment (antoine-de) [inline fabfile/component/tyr.py]: just env.use_systemd no ?

File: fabfile/env/platforms.py [hunk 3]
```diff
@@ -43,6 +43,9 @@
 # local path to navitia packages
 env.debian_packages_path = None
 
+# debian8 using systemd
+env.use_systemd = None
```
Comment (antoine-de) [inline fabfile/env/platforms.py]: False no ?

File: fabfile/component/tyr.py [hunk 4]
```diff
@@ -118,8 +118,13 @@ def setup_tyr():
...
+        _upload_template('tyr/systemd_tyr_worker.jinja', env.tyr_worker_service_file,
+                     user='root', mode='644', context={'env': env})
+    else:
+        _upload_template('tyr/tyr_worker.jinja', env.tyr_worker_service_file,
```
Comment (kinnou02) [inline fabfile/component/tyr.py]: we should'nt upload the file to the same place no?

**Comment by kinnou02** [comment]
we also should do this for all services

File: fabfile/component/tyr.py [hunk 5]
```diff
@@ -154,8 +159,13 @@ def upgrade_tyr_packages():
     #we want the version of the system for these packages
     run('''sed -e "/protobuf/d" -e "/psycopg2/d"  /usr/share/tyr/requirements.txt > /tmp/tyr_requirements.txt''')
     require.python.install_requirements('/tmp/tyr_requirements.txt', use_sudo=True, exists_action='w', upgrade=True)
-    _upload_template('tyr/tyr_worker.jinja', env.tyr_worker_service_file,
+    if env.distrib == 'debian8' and env.use_systemd:
```
Comment (TeXitoi) [inline fabfile/component/tyr.py]: The distrib test seems bad, we want to install that on all systemd systems, not only debian8

File: fabfile/component/tyr.py [hunk 6]
```diff
@@ -180,7 +190,11 @@ def upgrade_cities_db():
 @roles('tyr_master')
 def setup_tyr_master():
     utils.require_directory(env.ed_basedir, owner='www-data', group='www-data', use_sudo=True)
-    _upload_template('tyr/tyr_beat.jinja', env.tyr_beat_service_file,
+    if env.distrib == 'debian8' and env.use_systemd:
+        _upload_template('tyr/systemd_tyr_beat.jinja', env.tyr_beat_service_file,
```
Comment (TeXitoi) [inline fabfile/component/tyr.py]: It must be installed in  /etc/systemd/system/{service_name}.service

See https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System_Administrators_Guide/sect-Managing_Services_with_systemd-Unit_Files.html

File: fabfile/component/tyr.py [hunk 7]
```diff
@@ -180,7 +190,11 @@ def upgrade_cities_db():
 @roles('tyr_master')
 def setup_tyr_master():
     utils.require_directory(env.ed_basedir, owner='www-data', group='www-data', use_sudo=True)
-    _upload_template('tyr/tyr_beat.jinja', env.tyr_beat_service_file,
+    if env.distrib == 'debian8' and env.use_systemd:
```
Comment (TeXitoi) [inline fabfile/component/tyr.py]: same here

File: fabfile/component/tyr.py (hunk 6 above)
Comment (Hoshiyo) [inline fabfile/component/tyr.py]: Yes it is specify on CanalTP/navitia_deployment_conf#103

**Comment by Hoshiyo** [comment]
Yes I will create for kraken

**Comment by Hoshiyo** [comment]
@antoine-de I changed the init path with the comment of @kinnou02 to CanalTP/navitia_deployment_conf#103

File: fabfile/component/kraken.py [hunk 8]
```diff
@@ -479,12 +489,12 @@ def remove_kraken_instance(instance, purge_logs=False):
     """
     instance = get_real_instance(instance)
 
-    sudo("service kraken_%s stop; sleep 3" % instance.name)
+    sudo("service kraken_{} stop; sleep 3".format(instance.name))
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: :+1:

File: fabfile/env/platforms.py [hunk 9]
```diff
@@ -43,6 +43,12 @@
 # local path to navitia packages
 env.debian_packages_path = None
 
+# debian8 using systemd
+env.use_systemd = False
+
+# init path
+env.init_path = '/etc/init.d/'
```
Comment (antoine-de) [inline fabfile/env/platforms.py]: I think it would be nice for init_path to be a method to compute is with env.use_systemd no ?

**Comment by Hoshiyo** [comment]
@antoine-de I need to specify the second arg of env.tyr_beat_service_file in CanalTP/navitia_deployment_conf#103
Can I have a best solution?

**Comment by antoine-de** [comment]
@Hoshiyo can you tell me when the PR is amended ?

**Comment by Hoshiyo** [comment]
@antoine-de I'm waiting the check

**Comment by antoine-de** [comment]
OK for me

**Pull request merged**